arq/connections.py

```diff
@@ -129,14 +143,14 @@ async def enqueue_job(
         defer_by_ms = to_ms(_defer_by)
         expires_ms = to_ms(_expires)
 
-        with await self as conn:
+        async with self as conn:
             pipe = conn.pipeline()
-            pipe.unwatch()
-            pipe.watch(job_key)
-            job_exists = pipe.exists(job_key)
-            job_result_exists = pipe.exists(result_key_prefix + job_id)
-            await pipe.execute()
-            if await job_exists or await job_result_exists:
+            await pipe.unwatch()
```

Comment:
Maybe you can remove `unwatch`? It was required since with aioredis<2 connections often still had a watch in place from previous commands

Reply:
Okay, ran tests several times, seems to be working fine without unwatch.